working/0546-patterns/patterns-feature-specification.md

```diff
@@ -1227,69 +1347,90 @@ var (a, int b) = ... // Schema is `(?, int)`.
 
 The context type schema for a pattern `p` is:
 
-*   **List binder or matcher**: A type schema `List<E>` where:
-    *   If `p` has a type argument, then `E` is the type argument.
-    *   Else `E` is the greatest lower bound of the type schemas of all element
-        subpatterns.
-
-*   **Map binder or matcher**: A type schema `Map<K, V>` where:
-    *   If `p` has type arguments then `K`, and `V` are those type arguments.
-    *   Else `K` is the least upper bound of the types of all key expressions
-        and `V` is the greatest lower bound of the context type schemas of all
-        value subpatterns.
+*   **Or**: The least upper bound of the context type schemas of the branches.
 
-*   **Record binder or matcher**: A record type scheme with positional and named
-    fields corresponding to the type schemas of the corresponding field
-    subpatterns. If a named field uses the shorthand syntax to infer a variable
-    subpattern with the same name as the field, then the type schema is `?` for
-    that field.
+*   **And**: The greatest lower bound of the context type schemas of the branches.
```

Comment:
Yeah, I think you'd get `Never`. Basically, I defer to what our existing spec for GLB says here. :)

I don't think having `Never` appear here is problematic, but I'm definitely not sure. It won't do much as far as type inference goes, but that's OK. It could still be reasonable to allow it, for example:

```dart
class A {}
class B {}

class C implements A, B {}

main(Object obj) {
  if (var (A a & B b) = obj) {
    // Can use a and b here...
  }
}
```